### ISSUE TITLE
Please Collaborate with FrostWire

### ISSUE THREAD

**Issue by gubatron** [bitcoin-torrent/bitcoin-torrent#1]
[We](http://www.frostwire.com/) are OpenSource, and we've been thinking a great deal on how to do this.

http://github.com/frostwire/frostwire-common
http://github.com/frostwire/frostwire-desktop
http://github.com/frostwire/frostwire-jlibtorrent
http://github.com/frostwire/frostwire-android

We already have done many of your TO-DO's and have millions of users worldwide on both desktops and android, we use [frostwire-jlibtorrent](http://github.com/frostwire/frostwire-jlibtorrent) (A libtorrent wrapper library we made for java) for bittorrent integration and could use bitcoinj or a web api for bitcoin integration.

Make file transfer fast - done
Multiple files support, not only 1 file - done
DHT support - done

A few weeks ago [we posted this video](https://www.youtube.com/watch?v=W-0WJLpAenw), looking to bounce ideas with people that knew more about the Bitcoin protocol capabilities than we do. Many pointed us to Micropayments, other pointed us to BIP70 and using pre-signed refund transactions (still trying to wrap our heads around this)

Our proposal for this however is meant to let the content creator monetize and share the revenue with the seeders.

When creating a paid for torrent, the content creator gets to decide the following rev. split which will be saved on the torrent metadata (calculated in the torrent infohash)

**Use case example**

Say you've made a music album and you want to sell it for $5.
You could keep 95% of the download and give 4% to seeders and 1% for bitcoin transaction fees.
Now supose the torrent for such album has about 12 files, which are cut into 250 torrent pieces.

What we want to do is to have a transaction on which the downloader will pay one of many seeders per piece, and the payment will be split 95% to the bitcoin address of the content creator, 4% to the seeder giving the piece and 1% of that piece going to miners.

We want to include the miners because we foresee a large number of transactions, just one downloader, 250 transactions, although maybe with the micropayment channel things would be different.

We are stuck at protecting all parties, the seeder from not giving a free piece, the downloader from not getting a fake piece, and the content creator getting paid for a legitimate torrent download.

**What we're not trying to do**
We're not trying to make this a piracy proof solution, we know this is impossible and easily circumvented by simply creating another torrent of the content that has been downloaded.

It's about trying to provide legal means to sell and purchase content on the bittorrent network so that you can sell uncensored content using a network proven to distribute content worldwide efficiently.

You can contact me on twitter @gubatron, or by email to contact@frostwire.com if you'd like to collaborate. In the meantime I'll be pouring over your implementation.

This is so amazing, thanks for sharing this

**Comment by alxhotel** [comment]
Awesome. I think this means: "case closed" :D

**Comment by EmergentBehavior** [comment]
I second working with @gubatron. Better to collaborate and integrate.

Also, Storj (which I am a dev for) [will be incorporating libtorrent](https://github.com/storj/storjtorrent) for file transfer so insinuating we don't use torrent tech in the README is incorrect. Also, SJCX does not have its own separate blockchain (is a Counterparty asset) and metadata will be effectively stored on Bitcoin blockchain via tiny hashes using Factom or something similar. 

Basically, we're not reinventing the wheel.

**Comment by ThomasdenH** [comment]
I think the transfer should begin with the seeder send a part, and then receiving a payment for it. That way, no one will lose money. To prevent someone abusing this system by downloading free parts from different peers, a reputation system could be built. Now we want to make it costly to make a new 'account', or at least make it costly to make many of them. This could be done by using something similar to the mining of bitcoin. One should create an identity that hashes to a value that fits a set of rules. If this takes for example 5-15 minutes to do, it would be perfectly feasible for the ordinary user, but to keep creating new identities for every couple of parts would be very costly. I think it then would be easier to find a download in another way.

**Comment by gubatron** [comment]
> "a reputation system could be built"

a [reputation plugin is being built](https://github.com/ssiloti/bep-persistent-credit/blob/master/one-hop-rep.rst) currently by one of the top libtorrent's contributors, however it's not meant for cryptocurrency. decentralized reputation is a tough problem, I'm waiting to see his final submission to see if it could be used for purposes similar to the ones you propose @ThomasdenH 

There's also JoyStream which was working with the libtorrent mailing list, his code is closed for now, he said he will be opening it. He built the entire thing with payment channels, and I'm very curious to know how he handles multiple payment channels on the swarm. His vision was a lot different than mine, and it lends itself more to criminal copyright infringement. perhaps he'll be adjusting his implementation to be more like what we want to propose for content creators.
